### PR TITLE
Issue #7624: Adds examples to doc for NoWhiteSpaceAfter

### PR DESCRIPTION
**Pull request by Perin-Mangukiya**
Issue #7624 
Added examples for 3 tokens DOT, ARRAY_DECLARATOR and ARRAY_INIT

### CODE REVIEW

**Comment by nmancus1** [comment]
Github, generate site

**Review by nmancus1 — CHANGES_REQUESTED** [review]
Let's do three configs with examples: 

```xml

<module name="NoWhitespaceAfter"/>

```

This is default (already exists in documentation) but has no examples. Please add example with all default tokens, violations and good formatting, and some with line breaks around tokens to give user a good idea about default check behavior.

```xml
<module name="NoWhitespaceAfter">
  <property name="tokens" value="DOT"/>
  <property name="allowLineBreaks" value="false"/>
</module>
```

This is a good way to show user exact behavior of allowLineBreaks = false.

```xml

<module name="NoWhitespaceAfter">
  <property name="tokens" value=" ARRAY_DECLARATOR, INDEX_OP"/>
</module>
```

Let's show user all the nuances of these two tokens, since they are treated differently than others (as stated by existing documentation).

Please remove other examples, I am not sure of value to the user to go through each token separately with and without line breaks.

**Comment by Perin-Mangukiya** [comment]
Ok, I will change it.

**Comment by Perin-Mangukiya** [comment]
@nmancus1 please help me with the above failing checks.

**Comment by nmancus1** [comment]
> please help me with the above failing checks.

Please share what you have tried and what you do not understand so that we can help.

**Comment by romani** [comment]
Problems:
```
[ERROR] [checkstyle] [ERROR] /drone/src/src/xdocs/config_whitespace.xml:1503: Line should not be longer than 100 symbols [lineLengthXml]
[ERROR] [checkstyle] [ERROR] /drone/src/src/xdocs/config_whitespace.xml:1521: Unnecessary consecutive lines [noConsecutiveLines]
```

Please fix such violation.
Before push please make sure that "mvn verify" is passing, if failing - read message and try to fix.

**Comment by Perin-Mangukiya** [comment]
Please review it.

**Comment by romani** [comment]
Github, generate site

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java [hunk 1]
```diff
@@ -91,14 +91,111 @@
...
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * class Test2 {
+ *   public void function1() {}
+ *   public void function2() {}
+ * }
+ *
+ * class Test extends Test2 {
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java:104]: Please remove Test2
And override from https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html
toString() and finalize()

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java [hunk 2]
```diff
@@ -91,14 +91,111 @@
...
+ *   int a, b;
+ *   Test(int a, int b) {
+ *     this.
+ *         a = a; // violation, linebreak after '.' is not allowed
+ *     this.b = b; // OK
+ *   }
+ *
+ *   public void function() {}
+ *
+ *   public static void main(String[] args) {
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java:166]: Please make code block same for all configs.
To show user how behavior on the same code is changing if you change properties of Check..

**Review by romani — CHANGES_REQUESTED** [review]
Items:

**Comment by nick-mancuso** [comment]
@Perin-Mangukiya ping

**Comment by Perin-Mangukiya** [comment]
I will do it. Now my exams are going on so I am not active, in next week I will complete it.

**Comment by Perin-Mangukiya** [comment]
Sorry! By mistake I closed my PR.

**Comment by romani** [comment]
@Perin-Mangukiya , please address review items and reply "done" to each of them.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java (hunk 2 above)
Comment (Perin-Mangukiya) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java:166]: done

**Comment by romani** [comment]
@Perin-Mangukiya, please keep only one commit in PR, rebase properly.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java (hunk 1 above)
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java:104]: I do not see it done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java (hunk 2 above)
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java:166]: Not done

**Pull request closed without merging**